include/mbgl/map/map_observer.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+namespace mbgl {
+
+class MapObserver {
+public:
+    static MapObserver& nullObserver() {
+        static MapObserver mapObserver;
+        return mapObserver;
+    }
+
+    virtual void onRegionWillChange() {}
+    virtual void onRegionWillChangeAnimated() {}
```

Comment:
I think so - I'm adding two enums to `MapObserver`: `CameraChangeMode` and `RenderMode`.